dora/src/bytecode/generator.rs

```diff
@@ -388,24 +390,47 @@ impl<'a, 'ast> AstBytecodeGen<'a, 'ast> {
             Register::zero()
         };
         self.gen.set_position(expr.pos);
+
         let arg_start_reg = match *call_type {
-            CallType::CtorNew(ty, _) => {
+            CallType::CtorNew(_, _) | CallType::Method(_, _, _) => 1,
+            _ => 0,
+        };
+
+        for (idx, arg) in expr.args.iter().enumerate() {
+            let arg_reg = start_reg.offset(idx + arg_start_reg);
+            self.visit_expr(arg, DataDest::Reg(arg_reg));
+        }
+
+        match *call_type {
+            CallType::CtorNew(_, _) => {
+                let ty = fct.params_with_self()[0];
```

Comment:
I think you could take the object type out of `CtorNew` as well.

Reply:
Then you wouldn't need `specialize_for_call_type` here as well, since this type should already be specialized for that.

Reply:
Before my change it was taken out of `CtorNew` and I'm pretty sure that it didn't work for classes with generics. But I will look into it again.

Reply:
I stand corrected, `CtorNew` has the correct type. Changed it back.